File: Beelina.APP/src/app/_models/product-stock-audit-item.ts
import { StockAuditSourceEnum } from '../_enum/stock-audit-source.enum';
import { DateFormatter } from '../_helpers/formatters/date-formatter.helper';
import { IModelNode } from '../_interfaces/imodel-node';
import { Entity } from './entity.model';

export class ProductStockAuditItem extends Entity implements IModelNode {
  public transactionNumber: number
  public quantity: number;
  public stockAuditSource: StockAuditSourceEnum;
  public modifiedDate: Date;
  public modifiedBy: string;

  get dateCreatedFormatted(): string {
    return DateFormatter.format(this.modifiedDate, 'MMM DD, YYYY HH:mm');
  }

  constructor() {
    super();
  }
}

File: Beelina.LIB/Models/ProductStockAuditItem.cs
using Beelina.LIB.Enums;

namespace Beelina.LIB.Models
{
    public class ProductStockAuditItem
    {
        public int Id { get; set; } // Product Stock Id or Transaction Id
        public string TransactionNumber { get; set; } // Either Withdrawal Slip No or Transaction No.
        public string PlateNo { get; set; }
        public int Quantity { get; set; }
        public StockAuditSourceEnum StockAuditSource { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime ModifiedDate { get; set; }
    }
}
